EpikCord/ext/tasks.py

```diff
@@ -0,0 +1,82 @@
+import datetime
+from typing import Callable, Optional
+import asyncio
+from ..exceptions import TaskFailedError
+
+_EPOCH = datetime.datetime(1970, 1, 1)
+__all__ = (
+    "TimeParser",
+    "Tasks",
+)
+
+
+
+
+@staticmethod
+def total_seconds(
+    day: int = 0,
+    hour: int = 0,
+    minute: int = 0,
+    seconds: float = 0,
+) -> float:
+    hour += day * 24
+    minute += hour * 60
+    seconds += minute * 60
+    return seconds
+
+
+class Tasks:
+    def __init__(self, client) -> None:
+        self.client = client
+
+    def add_task(
+        self,
+        task: Callable[..., None],
+        interval: Optional[int] = None,
+        *args,
+        **kwargs,
+    ):
+        """Adds a background task (Tasks that run silently in the background)
+
+        This task can be used to monitor the bot's performance, change and send stuff routinely
+        through webhooks, etc
+        You can set the time of when it starts, stops and  add which arguments to give to the task
+        You may also define whether to run once or run indefinitely
+  
+        :param task: The function to run on the background
+        :type task: Callable[..., None]
+        :param interval:The interval (in seconds) for a delay between when the instance of the task stops and starts again.Defaults to 5.
+                        You may use `TimeParser.total_seconds` method, defaults to None
+        :type interval: Optional[int]
+        :param limit: The number of instances to start. Usually unlimited
+        :type limit: Optional[int] 
+        """        
+        kwargs["interval"] = interval
+
+        async def full_task(client, task, *args,**kwargs):
+            async def task_func(interval):
+                try:
+                    await task(client, *args,**kwargs)
+                except Exception as e:
+                    raise TaskFailedError(f"Task failed due to: {e}")
+
+                await asyncio.sleep(interval)
+
+            interval = interval if interval <= 0 else 5
+            task_start = False
+            nb_instances = kwargs.get("limit")
```

Comment:
What does `nb_instances` mean?

Reply:
done

Reply:
I just realized the var was useless, lemme remove it